agent/unit-test/softwarecontaineragent_unittest.cpp

```diff
@@ -0,0 +1,213 @@
+    }
+
+    std::shared_ptr<CommandJob> createCommandJob(const std::string &command)
+    {
+        log_debug() << "run command : " << command;
+        return nullptr;
+    }
+
+    void shutdown(unsigned int timeout __attribute__((unused)))
+    {
+        log_debug() << "Shutting Down.";
+    }
+
+    void suspend()
+    {
+        log_debug() << "Suspending.";
+    }
+
+    void resume()
+    {
+        log_debug() << "Resume.";
+    }
+
+    ReturnCode bindMount(const std::string &pathOnHost __attribute__((unused)),
+                         const std::string &pathInContainer __attribute__((unused)),
+                         bool readonly __attribute__((unused)))
+    {
+        return ReturnCode::SUCCESS;
+    }
+
+    bool previouslyConfigured()
+    {
+        return true;
+    }
+
+};
+
+class TestFactory: public SoftwareContainerFactory
+{
+public:
+    std::shared_ptr<ContainerAgentInterface> createContainer
+    (const ContainerID id __attribute__((unused)), std::unique_ptr<const SoftwareContainerConfig> config __attribute__((unused)))
+     {
+        std::shared_ptr<ContainerAgentInterface> container = std::shared_ptr<ContainerAgentInterface> (new TestContainerInterface());
+        return container;
+     }
+};
+} //namespace
+
+class SoftwareContainerAgentTest: public ::testing::Test
+{
+public:
+    SoftwareContainerAgent *sca;
+
+    const std::string configString = "[SoftwareContainer]\n"
+#ifdef ENABLE_NETWORKGATEWAY
+                                     "create-bridge = true\n"
+                                     "bridge-device = " + std::string(BRIDGE_DEVICE_TESTING) + "\n"
+                                     "bridge-ip = " + std::string(BRIDGE_IP_TESTING) + "\n"
+                                     "bridge-netmask = " + std::string(BRIDGE_NETMASK_TESTING) + "\n"
+                                     "bridge-netmask-bitlength = " + std::string(BRIDGE_NETMASK_BITLENGTH_TESTING) +"\n"
+                                     "bridge-netaddr = " + std::string(BRIDGE_NETADDR_TESTING) + "\n"
+#endif
+                                     "shutdown-timeout = 1\n"
+                                     "shared-mounts-dir = " + std::string(SHARED_MOUNTS_DIR_TESTING) + "\n"
+                                     "deprecated-lxc-config-path = " + std::string(LXC_CONFIG_PATH_TESTING) + "\n"
+                                     "service-manifest-dir = " + std::string(SERVICE_MANIFEST_DIR_TESTING) + "\n"
+                                     "default-service-manifest-dir = " + std::string(DEFAULT_SERVICE_MANIFEST_DIR_TESTING) + "\n";
+
+    const std::string valid_config = "[{\"enableWriteBuffer\": false}]";
+
+    void SetUp() override
+    {
+        std::unique_ptr<ConfigLoader> loader(new StringConfigLoader(configString));
+        std::unique_ptr<ConfigSource> mainConfig(new MainConfigSource(std::move(loader),
+                                                                      ConfigDefinition::typeMap()));
```

Comment:
I think TypeMap is a type and typeMap is an object to that type.  And yes it compiles.